### PR TITLE
[3.3.5] Core/Scripts: move npc_braug_dimspirit to SAI

### PR DESCRIPTION
**Pull request by ghost**
**Changes proposed:**

Move the core script `npc_braug_dimspirit` to Smart AI.

Additional change: The answers have been changed from containing only names
to include the text `is my answer.` from the matching lines in `broadcast_text`.
- added DB creature_text entries for Braug Dimspirit
- added DB gossip_menu_option entries for this NPC
- implemented NPC `Say` line in reply to the wrong answers
- avoiding gossip exploits by turning off npcflag 1+2 in script started by clicking wrong gossip option
- added `action_param2`= 64 `SMARTCAST_COMBAT_MOVE` for casting spell [`Test of Lore`](http://www.wowhead.com/spell=6766/test-of-lore)
- added conditions for showing gossip depending on quest status
- removed deprecated SD comments in zone_stonetalon_mountains.cpp

**Target branch(es):** **3.3.5** - Quest removed in patch 4.0.3a (Cataclysm).

**Issues addressed:** Updates https://github.com/velinath/TrinityCore/issues/2 , [**Gossip migration from core scripts to SAI/DB ToDo List**](https://github.com/velinath/TrinityCore/issues/2)

**Tests performed:** (Does it build, tested in-game, etc)
- Builds OK, script removal & DB changes.
- Tested and found to be working as intended.

**Known issues and ToDo list:**
- [X] None that I know about.

(If you have found anything worth changing in this PR, please tell me.)

### CODE REVIEW

File: sql/updates/world/3.3.5/2016_09_32_17853_world_335.sql [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+(@Braug, 1,0,'Neltharion, hmm? As good an answer as any, yes?',       12,0,100,1,3000,0,8386,0, 'Braug Dimspirit - Test of Lore'),
+(@Braug, 2,0,'Alexstrazsa, the Life-Binder. A magnificent creature.', 12,0,100,1,3000,0,8387,0, 'Braug Dimspirit - Test of Lore'),
+(@Braug, 3,0,'Malygos, is it?',                                       12,0,100,1,3000,0,8388,0, 'Braug Dimspirit - Test of Lore'),
+(@Braug, 4,0,'That is the wrong answer, $n. Be off with you.',        12,0,100,1,3000,0,8393,0, 'Braug Dimspirit - Test of Lore');
+
+DELETE FROM `gossip_menu_option` WHERE `menu_id` = @menu;
+INSERT INTO `gossip_menu_option` (`menu_id`,`id`,`option_icon`,`option_text`,`OptionBroadcastTextID`,`option_id`,`npc_option_npcflag`,`action_menu_id`,`action_poi_id`,`box_coded`,`box_money`,`box_text`,`BoxBroadcastTextID`,`VerifiedBuild`) VALUES
+(@menu, 0, 0, 'Ysera is my answer.',        8381, 1, 1, 0, 0, 0, 0, '', 0, 0),
+(@menu, 1, 0, 'Neltharion is  my answer.',  8383, 1, 1, 0, 0, 0, 0, '', 0, 0),
+(@menu, 2, 0, 'Alexstrasza is  my answer.', 8384, 1, 1, 0, 0, 0, 0, '', 0, 0),
```
Comment (Pitcrawler) [inline sql/updates/world/3.3.5/2016_09_32_17853_world_335.sql]: Is there an additional whitespace in this line and the one above?

Comment (tkrokli) [inline sql/updates/world/3.3.5/2016_09_32_17853_world_335.sql]: Yes, that is a correct observation. :-) But that is "unfortunately" also what is in `broadcast_text` :
```sql
SELECT * FROM `broadcast_text` WHERE `ID` IN (8381,8383,8384,8385);
```
```
  ID  Language  MaleText                    FemaleText                  EmoteID0  EmoteID1  EmoteID2  EmoteDelay0  EmoteDelay1  EmoteDelay2  SoundId    Unk1    Unk2  VerifiedBuild
----  --------  --------------------------  --------------------------  --------  --------  --------  -----------  -----------  -----------  -------  ------  ------  -------------
8381         0  Ysera is my answer.         Ysera is my answer.                0         0         0            0            0            0        0       0       1          18019
8383         0  Neltharion is  my answer.   Neltharion is  my answer.          0         0         0            0            0            0        0       0       1          18019
8384         0  Alexstrasza is  my answer.  Alexstrasza is  my answer.         0         0         0            0            0            0        0       0       1          18019
8385         0  Malygos is my answer.       Malygos is my answer.              0         0         0            0            0            0        0       0       1          18019
```
As you probably know, most/all text rows with reference to `broadcast_text` will use the `broadcast_text` content. 
If you think it is worth while changing the content in `broadcast_text`.`ID` 8383+8384, I will gladly change it here. ;-)

Comment (SnapperRy) [inline sql/updates/world/3.3.5/2016_09_32_17853_world_335.sql]: @Pitcrawler Blizzard is quite crazy when it comes to double spaces. Sometimes quests have them, sometimes gossips have them, sometimes *any kind of text* have them.

Not surprising that they made two gossip options with them and two without.

**Comment by Aokromes** [comment]
what's missing here?

**Comment by tkrokli** [comment]
I need to run a final test to make sure I nothing has changed and to see if there is anything else I can think of as a possible way to bug or exploit this quest. Completing this tonight, posting again when ready.

File: sql/updates/world/3.3.5/2016_10_32_17853_world_335.sql [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+(@Braug, 4,0,'That is the wrong answer, $n. Be off with you.',        12,0,100,1,3000,0,8393,0, 'Braug Dimspirit - Test of Lore');
+
+DELETE FROM `gossip_menu_option` WHERE `menu_id` = @menu;
+INSERT INTO `gossip_menu_option` (`menu_id`,`id`,`option_icon`,`option_text`,`OptionBroadcastTextID`,`option_id`,`npc_option_npcflag`,`action_menu_id`,`action_poi_id`,`box_coded`,`box_money`,`box_text`,`BoxBroadcastTextID`,`VerifiedBuild`) VALUES
+(@menu, 0, 0, 'Ysera is my answer.',        8381, 1, 1, 0, 0, 0, 0, '', 0, 0),
+(@menu, 1, 0, 'Neltharion is  my answer.',  8383, 1, 1, 0, 0, 0, 0, '', 0, 0),
+(@menu, 2, 0, 'Alexstrasza is  my answer.', 8384, 1, 1, 0, 0, 0, 0, '', 0, 0),
+(@menu, 3, 0, 'Malygos is my answer.',      8385, 1, 1, 0, 0, 0, 0, '', 0, 0);
+
+DELETE FROM `smart_scripts` WHERE `entryorguid` = @Braug AND `source_type` = 0;
```
Comment (Killyana) [inline sql/updates/world/3.3.5/2016_10_32_17853_world_335.sql]: missing:
``` sql
DELETE FROM `smart_scripts` WHERE `entryorguid` = @Script AND `source_type` = 9;
```

Comment (tkrokli) [inline sql/updates/world/3.3.5/2016_10_32_17853_world_335.sql]: Thank you, I added the script part (9) but forgot that it needs the `DELETE FROM` too.

**Pull request merged**